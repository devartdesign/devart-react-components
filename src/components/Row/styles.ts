import { StyleSheet } from 'aphrodite/no-important';

export default StyleSheet.create({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    marginRight: -15,
    marginLeft: -15,
    boxSizing: 'border-box'
  }
});
