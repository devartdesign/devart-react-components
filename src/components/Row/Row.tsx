import React, { memo, CSSProperties } from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export interface IRowProps {
  className?: string;
  style?: CSSProperties;
}

const Row = ({ children, className = '', style = {} }: React.PropsWithChildren<IRowProps>) => {
  return (
    <div key="Row" className={`${css(styles.container)} ${className}`.trim()} style={style}>
      {children}
    </div>
  );
};

export default memo(Row);
