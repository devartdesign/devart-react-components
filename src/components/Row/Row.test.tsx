import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';

import Row from './Row';

describe('Row Component', () => {
  global.console.error = () => null;
  let Component;
  it('should render', () => {
    Component = mount(<Row>data</Row>);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render with extra css classes', () => {
    Component = mount(
      <Row className="another-css-class">
        <p>data</p>
      </Row>
    );
    expect(create(Component).toJSON()).toMatchSnapshot();
  });
});
