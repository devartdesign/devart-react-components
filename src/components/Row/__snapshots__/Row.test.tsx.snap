// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component should render 1`] = `
.container {
  -webkit-box-lines: multiple;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

<div
  className="container"
  style={Object {}}
>
  data
</div>
`;

exports[`Row Component should render with extra css classes 1`] = `
.container {
  -webkit-box-lines: multiple;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

<div
  className="container another-css-class"
  style={Object {}}
>
  <p>
    data
  </p>
</div>
`;
